src/test/ui/lint/implicit_unsafe_autoref.rs

```diff
@@ -0,0 +1,14 @@
+// run-rustfix
+use std::ptr::{addr_of, addr_of_mut};
+
+unsafe fn _test_mut(ptr: *mut [u8]) -> *mut [u8] {
+    addr_of_mut!((*ptr)[..16])
+    //~^ error: implicit auto-ref creates a reference to a dereference of a raw pointer
+}
+
+unsafe fn _test_const(ptr: *const [u8]) -> *const [u8] {
+    addr_of!((*ptr)[..16])
+    //~^ error: implicit auto-ref creates a reference to a dereference of a raw pointer
+}
```

Comment:
Could you also add a test for the issue in https://github.com/rust-lang/rust/issues/73987 ?
And probably mark that issue also as being resolved by this PR.

Reply:
Currently the lint is not firing at that issue, I think it's because the lint checks for `implicit_ref!(*ptr)`, but that issue has `implicit_ref!((*ptr).field)` which doesn't match the pattern. I'll look into this the next week :)

Reply:
Please also add tests for `addr_of!((*ptr).field[..16])` then.

Reply:
Added in af4ade4cf60d64b9c8406bc32b21a2caef865fdd